docs/signer.rst

```diff
@@ -3,12 +3,71 @@ Signer API
 
 .. currentmodule:: securesystemslib.signer
 
+At its core the signer API defines abstract interfaces to create and verify
+cryptographic signatures:
```

Comment:
I would love it if the first sentence was about the value/service provided to reader, not the mechanism.

```suggestion
The signer module provides a simple and unified cryptographic signing and signature verification API designed to work consistently regardless of the underlying signing technology. This is achieved with abstract interfaces backed by multiple implementations:  
```